include/openPMD/IO/ADIOS/ADIOS2PreloadAttributes.hpp

```diff
@@ -147,8 +149,18 @@ namespace detail
                 "[ADIOS2] Requested attribute not found: " + name );
         }
         AttributeLocation const & location = it->second;
-        if( location.dt != determineDatatype< T >() )
+        Datatype determinedDatatype = determineDatatype< T >();
+        if( std::is_same< T, signed char >::value )
+        {
+            // workaround: we use Datatype::CHAR to represent ADIOS2 signed char
+            // (ADIOS2 does not have chars with unspecified signed-ness
+            // anyway)
+            determinedDatatype = Datatype::CHAR;
+        }
+        if( location.dt != determinedDatatype )
         {
+            std::cout << "location.dt=" << location.dt <<
+                "\tdetermineDatatype=" << determineDatatype< T > () << std::endl;
```

Comment:
Is this a leftover debug message and should go into the error message?

Reply:
Yep, will merge that with the error message